spicy/toolchain/src/compiler/codegen/parser-builder.cc

```diff
@@ -658,7 +658,7 @@ struct ProductionVisitor : public production::Visitor {
 
         else if ( ! meta.isFieldProduction() ) {
             // Need to move position ahead.
-            if ( state().ncur ) {
+            if ( auto field = meta.field(); field && ! field->isSkip() && state().ncur ) {
```

Comment:
I'm not quite sure I follow why this is needed: what is difference between with and without skip that needs us ignore ncur in the former case but not the latter?

Reply:
Yeah, this is not great. The issue is that even with this patch I do not store enough information about the actual production of the equivalent non-`skip` field in the `Skip` production. I initially tried to instead compute the full `production::Meta` in [when processing the `Field`](https://github.com/zeek/spicy/blob/1825bdbaea723ab184b45687821e9bee137d8d65/spicy/toolchain/src/compiler/codegen/grammar-builder.cc#L125) (i.e., move the `isSkip` block after the production computation so it can pass its `production::Meta` into `Skip`), but this caused changes to production names as we'd potentially compute production names for the same `Field` twice.

Reply:
I added one commit, see what you think.

Reply:
Thanks, that seems much more coherent. I left a few questions below, mostly for me to get a better understanding.